packages/apollo-tools/src/buildServiceDefinition.ts

```diff
@@ -174,6 +183,15 @@ export function buildServiceDefinition(
       });
     }
 
+    const directiveExtensions = Object.values(directivesMap).flat();
+
+    if (directiveExtensions.length > 0) {
```

Comment:
I don't think we're necessarily _extending_ the schema with directives - this might make more sense to include where we first build the schema up above:

```javascript
let schema = buildASTSchema({
  kind: Kind.DOCUMENT,
  definitions: [...typeDefinitions, ...directiveDefinitions]
});
```

I'd like to hear @martijnwalraven 's opinion on this though as I may be overlooking something.

Reply:
@trevor-scheer Yep, this definitely makes sense, and it's even required if you're defining schema directives because you need those definitions when first constructing the schema. Regardless, `extendSchema` has to re-create the schema so it might be better to avoid that extra pass.